cmd/kubeadm/app/util/config/initconfiguration.go

```diff
@@ -200,6 +206,21 @@ func ConfigFileAndDefaultsToInternalConfig(cfgPath string, defaultversionedcfg *
 	return internalcfg, nil
 }
 
+// ConfigFileAndDefaultsToInternalConfig takes a path to a config file and a versioned configuration that can serve as the default config
+// If cfgPath is specified, defaultversionedcfg will always get overridden. Otherwise, the default config (often populated by flags) will be used.
+// Then the external, versioned configuration is defaulted and converted to the internal type.
+// Right thereafter, the configuration is defaulted again with dynamic values (like IP addresses of a machine, etc)
+// Lastly, the internal config is validated and returned.
+func ConfigFileAndDefaultsToInternalConfig(cfgPath string, defaultversionedcfg *kubeadmapiv1beta1.InitConfiguration) (*kubeadmapi.InitConfiguration, error) {
```

Comment:
What about renaming into `LoadOrDefaultInitConfiguration` or something similar (that uses `InitConfiguration`)?

Reply:
This is done by @bruceauyeung in a yet to be merged PR #68595 .
It's better if it goes in this one though as it's more tied to the subject at hand.
@fabriziopandini @bruceauyeung WDYT?